src/main/java/frc/robot/commands/ramp/RampFollow.java

```diff
@@ -0,0 +1,46 @@
+package frc.robot.commands.ramp;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Ramp;
+import frc.robot.utils.Alignable;
+import frc.robot.utils.AutoAlignment;
+
+import java.util.function.Supplier;
+
+public class RampFollow extends Command {
+    private final Ramp ramp;
+    private final Supplier<Alignable> alignableSupplier;
+    private final Supplier<Pose2d> pose2dSupplier;
+
+    public RampFollow(Ramp ramp, Supplier<Alignable> alignableSupplier, Supplier<Pose2d> pose2dSupplier) {
+        this.ramp = ramp;
+        this.alignableSupplier = alignableSupplier;
+        this.pose2dSupplier = pose2dSupplier;
+        addRequirements(ramp);
+    }
+
+    @Override
+    public void execute() {
+        Alignable alignableNow = alignableSupplier.get();
+        if (alignableNow != null){
+            double encoderValue = Ramp.angleToEncoder(
+                    new Rotation2d(Math.PI / 2)
+                            .minus(AutoAlignment.getYaw(alignableNow,
+                                    pose2dSupplier.get().getTranslation(),
+                                    0.5)
+                            ).getDegrees()
+            );
+            if (Constants.RAMP_DEBUG) SmartDashboard.putNumber("RAMP_TARGET_AUTO",encoderValue);
+            ramp.setRampPos(encoderValue);
+        }
+    }
+
+    @Override
+    public boolean isFinished() {
+        return alignableSupplier.get() == null;
```

Comment:
this runs the risk of not detecting the termination condition as the drivetrain may see another command set the alienable without going through null first. Better to have this be part of a race group that terminates this command, or give it a provider that tells it to finish.
And it probably needs a timeout too...

Reply:
added deadline command. Regarding the timeout, I did not like the idea that they could still be holding the button and the Command times out.

Reply:
In the happy scenario, they would be holding the button and the command times out, which is sad but potentially acceptable (though Bat-Chen disagrees). In the unhappy scenario the command continues to run due to a bug even after the button is released and the battery drains.